apps/judicial-system/web/src/utils/validate.spec.ts

```diff
@@ -0,0 +1,209 @@
+  })
+})
+
+describe('Validate email format', () => {
+  test('should not be valid if @ is missing', () => {
+    // Arrange
+    const invalidEmail = 'testATtest.is'
+
+    // Act
+    const validation = validate(invalidEmail, 'email-format')
+
+    // Assert
+    expect(validation.isValid).toEqual(false)
+    expect(validation.errorMessage).toEqual('Netfang ekki á réttu formi')
+  })
+
+  test('should not be valid if the ending is less than two characters', () => {
+    // Arrange
+    const invalidEmail = 'testATtest.i'
+
+    // Act
+    const validation = validate(invalidEmail, 'email-format')
+
+    // Assert
+    expect(validation.isValid).toEqual(false)
+    expect(validation.errorMessage).toEqual('Netfang ekki á réttu formi')
+  })
+
+  test('should be valid if email is empty', () => {
+    // Arrange
+
+    // Act
+    const validation = validate('', 'email-format')
+
+    // Assert
+    expect(validation.isValid).toEqual(true)
+  })
+
+  test('should be valid if email contains - and . characters', () => {
+    // Arrange
+    const validEmail = 'garfield.lasagne-lover@garfield.io'
+
+    // Act
+    const validation = validate(validEmail, 'email-format')
+
+    // Assert
+    expect(validation.isValid).toEqual(true)
+  })
+
+  test('should be valid if email is valid', () => {
+    // Arrange
+    const validEmail = 'garfield@garfield.io'
+
+    // Act
+    const validation = validate(validEmail, 'email-format')
+
+    // Assert
+    expect(validation.isValid).toEqual(true)
+  })
+})
+
+describe('Validate phonenumber format', () => {
+  test('should fail if not in correct form', () => {
+    // Arrange
+    const phonenumber = '00292'
+
+    // Act
+    const r = validate(phonenumber, 'phonenumber')
+
+    // Assert
+    expect(r.isValid).toEqual(false)
+    expect(r.errorMessage).toEqual('Dæmi: 555-5555')
+  })
+
+  test('should pass if in correct form', () => {
+    // Arrange
+    const phonenumber = '555-5555'
+
+    // Act
+    const r = validate(phonenumber, 'phonenumber')
+
+    // Assert
+    expect(r.isValid).toEqual(true)
+  })
+})
+
+describe('Validate court case number', () => {
+  test.each`
+    courtCaseNumber
+    ${'R-1/2019'}
+    ${'R-22/2022'}
+    ${'R-7536/1993'}
+    ${'R-333/3333'}
+  `(
+    'should pass when case as correct format $courtCaseNumber',
+    ({ courtCaseNumber }) => {
+      const result = validate(courtCaseNumber, 'court-case-number')
+      expect(result.isValid).toEqual(true)
+    },
+  )
+
+  test.each`
+    courtCaseNumber
+    ${'2019'}
+    ${'r-1/2019'}
+    ${'R.1/2019'}
+    ${'R/1/2019'}
+    ${'R/1-2019'}
+    ${'R/1-2019'}
+    ${'R-1-2019'}
+    ${'R-1/201'}
+    ${'R-1/201'}
+    ${'R-12345/2014'}
+  `(
+    'should fail if case number as wrong format $courtCaseNumber',
+    ({ courtCaseNumber }) => {
+      const result = validate(courtCaseNumber, 'court-case-number')
+      expect(result.isValid).toEqual(false)
+      expect(result.errorMessage).toEqual('Dæmi: R-1234/2022')
```

Comment:
Won't this test fail next year, because the year will change? Consider injecting the year or use some kind of date factory which can be mocked in the test.

Reply:
Ofc! 🤦‍♂️